crates/gazette/src/lib.rs

```diff
@@ -68,8 +70,59 @@ impl Error {
             Error::Parsing { .. } => false,
             Error::Protocol(_) => false,
             Error::UUID(_) => false,
+            Error::JWT(_) => false,
         }
     }
 }
 
 pub type Result<T> = std::result::Result<T, Error>;
+
+/// Dial a gRPC endpoint with opinionated defaults and
+/// support for TLS and Unix Domain Sockets.
+pub async fn dial_channel(endpoint: &str) -> Result<tonic::transport::Channel> {
```

Comment:
My assumption is that `endpoint` here will be the generic data plane address, and I just want to sanity check that this will create a new tcp connection each time (and that tonic will round-robin them), vs potentially re-using an existing one. Basically, I just want to make sure that agents will distribute the connector proxy requests across all the reactors in a data plane.

Reply:
:+1: yea that's certainly the intent.

`dial_channel` will start a new transport::Channel which is basically an H/2 connection over which many requests may be multiplexed (and Channel has a cheap Clone to allow multiple RPCs to be started).

As you say, we want to dial a new connection each time and lean on multiple A / AAAA records to load balancing across all available reactors. If we happen to block up on one reactor, the semaphore is intended to delay the RPC but prevent an OOM.